src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -166,129 +160,134 @@ class HierarchicalNSW : public VecsimBaseObject {
     idType getEntryPointId() const;
     labelType getExternalLabel(idType internal_id) const;
     VisitedNodesHandler *getVisitedList() const;
+    virtual VecSimIndexInfo info() const override;
+    virtual VecSimInfoIterator *infoIterator() const override;
+    virtual VecSimBatchIterator *newBatchIterator(const void *queryBlob,
+                                                  VecSimQueryParams *queryParams) override;
+    bool preferAdHocSearch(size_t subsetSize, size_t k, bool initial_check) override;
     char *getDataByInternalId(idType internal_id) const;
     inline linklistsizeint *get_linklist_at_level(idType internal_id, size_t level) const;
     unsigned short int getListCount(const linklistsizeint *ptr) const;
     void resizeIndex(size_t new_max_elements);
-    void removePoint(labelType label);
-    void addPoint(const void *data_point, labelType label);
-    dist_t getDistanceByLabelFromPoint(labelType label, const void *data_point);
+    int deleteVector(labelType label) override;
+    int addVector(const void *vector_data, labelType label) override;
+    double getDistanceFrom(labelType label, const void *vector_data) const override;
     idType searchBottomLayerEP(const void *query_data, void *timeoutCtx,
                                VecSimQueryResult_Code *rc) const;
-    vecsim_stl::max_priority_queue<dist_t, labelType>
-    searchKnn(const void *query_data, size_t k, void *timeoutCtx, VecSimQueryResult_Code *rc) const;
-    VecSimQueryResult *searchRange(const void *query_data, dist_t radius, void *timeoutCtx,
-                                   VecSimQueryResult_Code *rc) const;
-    bool isLabelExist(labelType label);
```

Comment:
Why did you remove this?

Reply:
Unnecessary. This function was added to check if a label exists in the index from the wrapper class, and now there is no wrapper. We also use the return value of the lookup when we look for a label so there is no point in looking up twice.